Add remix guide (#595)
onflow/docs: docs/evm/Build/Guides/Deploy Solidity Contract/deploy-solidity-contract-with-remix.md

---

> title: Using Re,ox

Reviewer: ```suggestion
title: Using Remix
```

Reviewer: Might rename directory menu to **Deploying a Smart Contract/**

description: "Using Remix to deploy a solidity contract to FlowEVM."
sidebar_label: Using Remix
sidebar_position: 1
---

# Using Remix

Remix is an open-source, web-based development environment tailored for EVM smart contract development. It provides developers with a comprehensive suite of tools to write, deploy, and test smart contracts in Solidity. More Information [Remix](https://remix.ethereum.org/)

## Add Flow EVM Network to metamask

<img src="./Remix-adding-metamask-network.gif" width="50%" alt="Add Flow EVM Network" />

> Navigate to [Using EVM](../../../using.mdx) page to click on the convenient button to add Flow EVM network information to your metamask.

Reviewer: ```suggestion
Navigate to [Using EVM](../../../using.mdx) and select the provided button to automatically add Flow EVM network details to your MetaMask wallet.
```

### Using Ethersjs to call HelloWorld smart contract

```javascript
// Import ethers from the ethers.js library
const { ethers } = require('ethers');

// Define the contract ABI
const contractABI = [
    "function hello() public view returns (string memory)"
];

// Define the contract address
const contractAddress = "0x8a120383e6057b1f3aef4fa9b89c2f1b0a695926";

// Connect to the Ethereum network
// This example uses the default provider from ethers.js, which connects to the Ethereum mainnet.
// For a testnet or custom RPC, use ethers.getDefaultProvider('networkName') or new ethers.providers.JsonRpcProvider(url)
  const provider = new ethers.providers.Web3Provider(window?.ethereum);
              

// Create a new contract instance
const contract = new ethers.Contract(contractAddress, contractABI, provider);

// Call the hello function of the contract
async function getGreeting() {
    const greeting = await contract.hello();
    console.log(greeting);
}

// Execute the function
getGreeting();

```

1. Create a new file under `scripts`
2. Paste in above JavaScript code
3. Click on green play button
4. Verify the greeting is "Hello World!"

> Use the steps below to change the greeting and retrieve the greeting again.

Reviewer: ```suggestion
Use the steps below to change the greeting and retrieve the updated greeting.
```

## Updating deployed Smart Contract

> <img src="./Remix-update-greeting.gif" width="50%" alt="Update Smart Contract">

Reviewer: needs closing <img> tag
